### DISCUSSION TITLE
zgui: ImGuizmo

### DISCUSSION THREAD

**Discussion by a-day-old-bagel** [zig-gamedev/zig-gamedev#611] [Ideas]
[ImGuizmo](https://github.com/CedricGuillemet/ImGuizmo/tree/ba662b119d64f9ab700bb2cd7b2781f9044f5565) is a helper for imgui that offers a 3D transform gizmo without any extra dependencies.

It seems to be widely used and popular, stable and tested, etc.

How likely is it that we can get this included in zgui? I'm happy to attempt any of the work involved, if it turns out to be something that everybody wants.

For clarity, the reason I think this would fit well into zig-gamedev is that many of the types of games (or game engines) that will use zig-gamedev are likely to include in-game editors of some form.

My own project has come to the point that I need an in-game editor, for example, and since I'm going to be adapting ImGuizmo for use in zig for this, I figured "why not suggest it become a contribution to zig-gamedev."

**Comment by hazeycode** [comment]
I think this would be a very nice addition to zgui and a warmly received contribution.

**Comment by a-day-old-bagel** [comment]
[Pull Request](https://github.com/zig-gamedev/zig-gamedev/pull/634)
